src/Microsoft.TemplateSearch.TemplateDiscovery/Program.cs

```diff
@@ -71,155 +48,102 @@ private static async Task Main(string[] args)
-                    }
-                    else
-                    {
-                        return false;
-                    }
-                }
-
-                else
-                {
-                    return false;
-                }
-            }
+                Description = "Include preview packs in the results (by default, preview packs are ignored and the latest stable pack is used.",
+            };
 
-            return true;
-        }
+            Option<int> pageSizeOption = new Option<int>("--pageSize", getDefaultValue: () => _defaultPageSize)
+            {
+                Description = "(debugging) The chunk size for interactions with the source.",
+            };
 
-        private static bool TryGetFlagValue(string[] args, int index, out string? value)
-        {
-            if (index < args.Length && !args[index + 1].StartsWith("-"))
+            Option<bool> onePageOption = new Option<bool>("--onePage")
+            {
+                Description = "(debugging) Only process one page of template packs.",
+            };
+
+            Option<bool> savePacksOption = new Option<bool>("--savePacks")
+            {
+                Description = "Don't delete downloaded candidate packs (by default, they're deleted at the end of a run).",
+            };
+
+            Option<bool> noTemplateJsonFilterOption = new Option<bool>("--noTemplateJsonFilter")
+            {
+                Description = "Don't prefilter packs that don't contain any template.json files (this filter is applied by default).",
+            };
+
+            Option<bool> verboseOption = new Option<bool>(new[] { "-v", "--verbose" })
+            {
+                Description = "Verbose output for template processing.",
+            };
+
+            Option<SupportedQueries[]> queriesOption = new Option<SupportedQueries[]>("--queries")
+            {
+                Arity = ArgumentArity.OneOrMore,
+                Description = $"The list of providers to run. Supported providers: {string.Join(",", Enum.GetValues<SupportedQueries>())}.",
+                AllowMultipleArgumentsPerToken = true,
+            };
+            queriesOption.FromAmong(Enum.GetValues<SupportedQueries>().Select(e => e.ToString()).ToArray());
+
+            Option<DirectoryInfo> packagesPathOption = (new Option<DirectoryInfo>("--packagesPath")
+            {
+                Description = $"Path to pre-downloaded packages. If specified, the packages won't be downloaded from NuGet.org.",
+            }.ExistingOnly());
+
+            var allOptions = new Option[]
+            {
+                basePathOption,
+                allowPreviewPacksOption,
+                pageSizeOption,
+                onePageOption,
+                savePacksOption,
+                noTemplateJsonFilterOption,
+                verboseOption,
+                queriesOption,
+                packagesPathOption
+            };
+
+            foreach (var option in allOptions)
```

Comment:
If you move the `RootCommand` instantiation down here, you can use a collection initializer and you don't need the loop, something like:

```cs
var rootCommand = new RootCommand("Generates the template package search cache file based on the packages available on NuGet.org.")
{
    basePathOption,
    allowPreviewPacksOption,
    pageSizeOption,
    onePageOption,
    savePacksOption,
    noTemplateJsonFilterOption,
    verboseOption,
    queriesOption,
    packagesPathOption
};
```